src/ui/veles_mainwindow.cc

```diff
@@ -839,6 +908,31 @@ bool MainWindowWithDetachableDockWidgets::event(QEvent* event) {
   return QMainWindow::event(event);
 }
 
+bool MainWindowWithDetachableDockWidgets::splitDockWidgetImpl(
+    QDockWidget* first, QDockWidget* second, Qt::Orientation orientation) {
+  auto tabified_docks = tabifiedDockWidgets(first);
+  for (auto dock_widget : tabified_docks) {
+    removeDockWidget(dock_widget);
+  }
+  tabified_docks.removeOne(second);
+
+  second->show();
+  splitDockWidget(first, second, orientation);
+
+  for (auto dock_widget : tabified_docks) {
+    dock_widget->show();
+    if (dock_widget != second) {
```

Comment:
How can dock_widget == second happen? You explicitly remove second a few lines up. Or can the splitDockWidget() append it back?